src/Action/GetShortObjectDescriptionAction.php

```diff
@@ -23,43 +25,73 @@
 final class GetShortObjectDescriptionAction
 {
     /**
-     * @var Pool
+     * @var AdminFetcherInterface
      */
-    private $pool;
+    private $adminFetcher;
 
     /**
      * @var Environment
      */
     private $twig;
 
-    public function __construct(Environment $twig, Pool $pool)
+    /**
+     * NEXT_MAJOR: Restrict second param to AdminFetcherInterface.
+     *
+     * @param Pool|AdminFetcherInterface $poolOrAdminFetcher
+     */
+    public function __construct(Environment $twig, $poolOrAdminFetcher)
     {
-        $this->pool = $pool;
         $this->twig = $twig;
+
+        if ($poolOrAdminFetcher instanceof AdminFetcherInterface) {
+            $this->adminFetcher = $poolOrAdminFetcher;
+        } elseif ($poolOrAdminFetcher instanceof Pool) {
+            @trigger_error(sprintf(
+                'Passing other type than %s in argument 2 to %s() is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will throw %s exception in 4.0.',
+                AdminFetcherInterface::class,
+                __METHOD__,
+                \TypeError::class
+            ), \E_USER_DEPRECATED);
+
+            $this->adminFetcher = new AdminFetcher($poolOrAdminFetcher);
+        } else {
+            throw new \TypeError(sprintf(
+                'Argument 2 passed to "%s()" must be either an instance of %s or %s, %s given.',
+                __METHOD__,
+                Pool::class,
+                AdminFetcherInterface::class,
+                \is_object($poolOrAdminFetcher) ? 'instance of "'.\get_class($poolOrAdminFetcher).'"' : '"'.\gettype($poolOrAdminFetcher).'"'
+            ));
+        }
     }
 
     /**
      * @throws NotFoundHttpException
      */
     public function __invoke(Request $request): Response
     {
-        $code = $request->get('code');
-        $objectId = $request->get('objectId');
-        $uniqid = $request->get('uniqid');
-        $linkParameters = $request->get('linkParameters', []);
+        // NEXT_MAJOR: Remove this BC-layer.
+        if (null === $request->get('_sonata_admin')) {
+            @trigger_error(
+                'Not passing "_sonata_admin" value in the request is deprecated since sonata-project/admin-bundle 3.x'
+                .' and will throw %s exception in 4.0.',
```

Comment:
Did you forget to add a `sprintf()` call?

Reply:
Fixed, the sprintf is not needed now.